ops/op2/dispatch_slow.rs

```diff
@@ -227,6 +235,34 @@ pub(crate) fn with_fn_args(
   )
 }
 
+pub(crate) fn with_required_check(
+  generator_state: &mut GeneratorState,
+  required: u8,
+) -> TokenStream {
+  generator_state.needs_scope = true;
+  let arguments_lit = if required > 1 {
+    "arguments"
+  } else {
+    "argument"
+  };
+  gs_quote!(generator_state(fn_args, scope, self_ty) =>
+    (if #fn_args.length() < #required as i32 {
+      let msg = format!(
+        "Failed to execute '{}' on '{}': {} {} required, but only {} present",
```

Comment:
two notes on this:
- `Failed to execute '{}' on '{}':` we will need to extract this out when we add more of the webidl conversion so we dont duplicate this. no action needed now, just noting for future.
- this is only applicable for methods. could you handle normal functions as well? ie https://github.com/denoland/deno/blob/main/ext/canvas/01_image.js#L158-L166
- Same applies for constructors, as those say `Failed to construct '{}':`

Reply:
- noted.
- ah didn't know that. this pr is only for object wraps but easy to have this on functions too.
- yup will fix that, ig it also needs to be 'Failed to construct {}.{}` for static methods